### PR TITLE
Add package post-deploy scripts in project definition file

### PR DESCRIPTION
**Pull request by sfc-gh-melnacouzi**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description

- Add post-deploy scripts in project definition file. This allows users to add a section in project definition file that will execute post syncing to the stage. Example:
```
definition_version: 1.1
native_app:
  name: myapp
  artifacts:
    - src: app/*
      dest: ./
  package:
    post_deploy:
      - sql_script: scripts/package_post_deploy1.sql
      - sql_script: scripts/package_post_deploy2.sql
```
- This will trigger during `snow app deploy` or `snow app run` in dev mode, where application package is actually created.
- This PR also includes fixing the translation from PDF V2 to V2. Instead of converting v2 scripts to old style package scripts, we convert them to this post-deploy scripts.

### Testing
* Unit tests added
* Integration tests for post-deploy updated (and cleaned up/refactored) to include both app and package post-deploy tests.
* Manual testing on Mac and Windows

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -24,6 +24,9 @@
 ## Deprecations
 
 ## New additions
+* Added support for `native_app.package.post_deploy` scripts in project definition file.
```
Comment (sfc-gh-gbloom) [inline RELEASE-NOTES.md:27]: How do we want to communicate this in relation to `native_app.package.scripts`? Should we mark `scripts` as deprecated and encourage users to use this instead? Should we allow users to specify both?

File: src/snowflake/cli/plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py [hunk 2]
```diff
@@ -106,10 +94,9 @@ def _pdf_v2_to_v1(v2_definition: DefinitionV20) -> DefinitionV11:
             "distribution"
         ] = app_package_definition.distribution
     if app_package_definition.meta and app_package_definition.meta.post_deploy:
-        pdfv1["native_app"]["package"]["scripts"] = [
-            _convert_v2_post_deploy_hook_to_v1_scripts(s)
-            for s in app_package_definition.meta.post_deploy
-        ]
+        pdfv1["native_app"]["package"][
+            "post_deploy"
+        ] = app_package_definition.meta.post_deploy
```
Comment (sfc-gh-gbloom) [inline src/snowflake/cli/plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py:99]: Shouldn't we convert both `scripts` and `post_deploy`? Users can still have `native_app.package.scripts` in their definition v1 file.

Comment (sfc-gh-melnacouzi) [reply src/snowflake/cli/plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py:99]: We are converting v2 to v1 here, so they can't have native_app.package.scripts in v2, or am I missing something?

Comment (sfc-gh-gbloom) [reply src/snowflake/cli/plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py:99]: Ah sorry, you're right, I got it backwards.

File: RELEASE-NOTES.md (hunk 1 above)
Comment (sfc-gh-melnacouzi) [reply RELEASE-NOTES.md:27]: I think that's a good point. We can mark in as deprecated in the documentation. 
I am not sure if we should put deprecation warning in the code yet...

Comment (sfc-gh-bdufour) [reply RELEASE-NOTES.md:27]: What's our story around users that specify both? We should be very clear about the ordering.

**Review by sfc-gh-astus — APPROVED** [review]

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -24,6 +24,10 @@
...
+  * These scripts will execute whenever a Native App Package is created or updated.
+  * Currently only supports SQL scripts: `post_deploy: [{sql_script: script.sql}]`
+  * Added deprecation warning for `native_app.package.scripts`
```
Comment (sfc-gh-gbloom) [inline RELEASE-NOTES.md:30]: Should this bullet move to be under the Deprecations section?

Comment (sfc-gh-melnacouzi) [reply RELEASE-NOTES.md]: moved

**Review by sfc-gh-gbloom — APPROVED** [review]

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**